spring/boot3-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/MappingWebEndpointPathMapper.java

```diff
@@ -32,30 +31,33 @@
 
 package com.linecorp.armeria.spring.actuate;
 
+import java.util.HashMap;
 import java.util.Map;
 
-import org.springframework.boot.actuate.endpoint.web.WebEndpointResponse;
-import org.springframework.boot.actuate.health.Status;
+import org.springframework.boot.actuate.endpoint.EndpointId;
+import org.springframework.boot.actuate.endpoint.web.PathMapper;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.util.StringUtils;
 
-import com.google.common.collect.ImmutableMap;
+import com.linecorp.armeria.common.annotation.Nullable;
 
 /**
- * Copied from spring-boot-actuator 2.3.0 to avoid breaking compatibility.
- * We used spring-boot-actuator's HealthStatusHttpMapper previously
- * but it has been deprecated since 2.2.0 and deleted since 2.3.0.
+ * Fork of <a href="https://github.com/spring-projects/spring-boot/blob/e3aac5913ed3caf53b34eb7750138a4ed6839549/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/endpoint/web/MappingWebEndpointPathMapper.java">MappingWebEndpointPathMapper</a>.
  */
-class SimpleHttpCodeStatusMapper {
-
-    private final Map<String, Integer> mappings;
+@Order(Ordered.HIGHEST_PRECEDENCE)
+class MappingWebEndpointPathMapper implements PathMapper {
+    private final Map<EndpointId, String> pathMapping;
 
-    SimpleHttpCodeStatusMapper() {
-        mappings = ImmutableMap.of(
-                Status.DOWN.getCode(), WebEndpointResponse.STATUS_SERVICE_UNAVAILABLE,
-                Status.OUT_OF_SERVICE.getCode(), WebEndpointResponse.STATUS_SERVICE_UNAVAILABLE
-        );
+    MappingWebEndpointPathMapper(Map<String, String> pathMapping) {
+        this.pathMapping = new HashMap<>();
+        pathMapping.forEach((id, path) -> this.pathMapping.put(EndpointId.fromPropertyValue(id), path));
     }
 
-    int getStatusCode(Status status) {
-        return mappings.getOrDefault(status.getCode(), WebEndpointResponse.STATUS_OK);
+    @Nullable
+    @Override
+    public String getRootPath(EndpointId endpointId) {
+        final String path = pathMapping.get(endpointId);
+        return StringUtils.hasText(path) ? path : null;
```

Comment:
Should we return it as is? The caller seems already checks it.
Otherwise, it would be better to normalize `path`s when `this.pathMapping` is created.